src/routes/snippets.rs

```diff
@@ -77,6 +77,30 @@ pub fn create_snippet(
     create_snippet_impl(&**storage, &snippet, origin.path())
 }
 
+#[allow(clippy::too_many_arguments)]
+#[get("/snippets?<title>&<syntax>&<tag>&<marker>&<limit>")]
+pub fn list_snippets(
+    storage: State<Box<dyn Storage>>,
+    title: Option<String>,
+    syntax: Option<String>,
+    tag: Option<String>,
+    marker: Option<String>,
+    limit: Option<usize>,
+    _content_type: NegotiatedContentType,
+    _user: BearerAuth,
+) -> Result<Output<Vec<Snippet>>, ApiError> {
+    let mut criteria = ListSnippetsQuery {
+        title,
+        syntax,
+        tags: tag.map(|v| vec![v]),
```

Comment:
Another way to write this would be to use the fact that `Option<T>` implements `Iterator`, so you could do something like: `tags: tag.iter().collect()`. I'm honestly not sure which one is better, but just wanted to point this out as an FYI :)

Reply:
Nice, I didn't know that. Another issue with `tag` is that I don't know how to collect multiple tag values from query parameters in Rocket. Taking into account that our python implementation always dealt with a single tag, I think it shouldn't be a problem for initial implementation of this route.

Reply:
> tag.iter().collect()

Unfortunately, it won't work since it produces `Option<Vec<&String>>` (vector over references) while we need vector over owned strings. I don't think it's worth to increase the chain of calls with cloning in order achieve iter-collect pattern here.

Reply:
> Nice, I didn't know that. Another issue with tag is that I don't know how to collect multiple tag values from query parameters in Rocket. Taking into account that our python implementation always dealt with a single tag, I think it shouldn't be a problem for initial implementation of this route.

Yeah, I completely forgot about that. Sure, a single tag sounds good!

> Unfortunately, it won't work since it produces Option<Vec<&String>> (vector over references) while we need vector over owned strings. I don't think it's worth to increase the chain of calls with cloning in order achieve iter-collect pattern here.

Ah, I confused `iter()` with `into_iter()`. The latter consumes its target and should give you an iterator over owned `String`s